codalab/client/local_bundle_client.py

```diff
@@ -928,7 +943,8 @@ def update_worksheet_items(self, worksheet_info, new_items):
         last_item_id = worksheet_info['last_item_id']
         length = len(worksheet_info['items'])
         worksheet = self.model.get_worksheet(worksheet_uuid, fetch_items=False)
-        check_worksheet_has_all_permission(self.model, self._current_user(), worksheet)
+        if not spec_util.is_dashboard(worksheet_info['name']):
```

Comment:
I think we should just create the dashboard under the first user who
accesses it, not root.  It's up to the admin of the local instance to visit
the dashboard first.

On Mon, Feb 8, 2016 at 10:17 PM, ChengshuLi <notifications@github.com>
wrote:

> In codalab/client/local_bundle_client.py
> <https://github.com/codalab/codalab-cli/pull/333#discussion_r52271132>:
>
> > @@ -928,7 +943,8 @@ def update_worksheet_items(self, worksheet_info, new_items):
> >          last_item_id = worksheet_info['last_item_id']
> >          length = len(worksheet_info['items'])
> >          worksheet = self.model.get_worksheet(worksheet_uuid, fetch_items=False)
> > -        check_worksheet_has_all_permission(self.model, self._current_user(), worksheet)
> > +        if not spec_util.is_dashboard(worksheet_info['name']):
>
> Yeah this is definitely not ideal. I put in these checks because the first
> user who visit the dashboard needs to create it as a root user, but he
> doesn't have the permission to populate the dashboard after creating it. So
> shall we make the creation of the dashboard part of the configuration
> process, not specific to any user? I think that might make more sense. If
> so, could you point me to some files that I should look at to achieve that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/codalab/codalab-cli/pull/333/files#r52271132>.
>